corehq/motech/dhis2/parse_response.py

```diff
@@ -0,0 +1,53 @@
+from operator import eq
+
+from jsonpath_rw import Child, Fields, Slice, Where, Root
+
+from corehq.motech.openmrs.jsonpath import Cmp
+
+
+def get_errors(response: dict) -> dict:
+    """
+    Searches response from DHIS2 for import errors, and returns a
+    dictionary of {jsonpath: error message}
+    """
+    if "response" in response:
+        response = response["response"]
+
+    errors = get_entity_errors(response)
+    errors.update(get_enrollments_errors(response))
+    errors.update(get_events_errors(response))
+    return errors
```

Comment:
Is there a risk of collisions using a dictionary here?  Would a list of tuples be more appropriate?

Reply:
The dictionary key is the JSON path to each error, which should be unique. But I'm not sure how DHIS2 reports multiple errors for the same value, and the documentation doesn't say. I'll test a bit more.

Reply:
It seems that when there are multiple problems with a payload, DHIS2 only reports the first error it finds. So, for better or worse, no risk of collisions.